Update rewards.md

acanidio-econ opened the pull request:
rewrite of the "buffer accounting" and "Solver's strategy" (plus a bunch of small things)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added CIP-36 to price-estimation guidance and expanded rewarded-quote conditions and examples.
  * Replaced “slippage” section with “Buffer accounting”: defines buffers, net slippage, weekly net settlement, and solver strategy impacts.
  * Clarified costs (vs. gas), negative payments (solver pays protocol), cap interactions (c_u), and noted cap binds in ≈9% of auctions (November 2025 timeframe).
  * Minor wording, formatting, and consistency edits throughout.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

acanidio-econ committed:
Update rewards.md

rewrite of the "buffer accounting" and "Solver's strategy"

acanidio-econ committed:
Update docs/cow-protocol/reference/core/auctions/rewards.md

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

acanidio-econ committed:
Update docs/cow-protocol/reference/core/auctions/rewards.md

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

fhenneke commented on docs/cow-protocol/reference/core/auctions/rewards.md:
```diff
@@ -32,48 +32,50 @@ Here $$\textrm{totalScore}$$ is the sum of the scores of all winning solutions i
 - BNB: $$c_l = 0.04 \;\textrm{BNB}$$, $$c_u = 0.048 \;\textrm{BNB}$$.
 
-Solutions with scores that are non-positive will be ignored. If only one solver submits solutions, $$\textrm{referenceScore}_i$$ is, by definition, zero. Formally, this corresponds to always considering the empty solution which does not settle any trades and has a score equal to zero as part of the submitted solutions.
+If only one solver submits solutions, $$\textrm{referenceScore}_i$$ is, by definition, zero. Formally, this corresponds to always considering the empty solution which does not settle any trades as part of the submitted solutions.
```
It does not help me much to explain an exceptional case by another exeptional case that is not even allowed in the protocol.
```suggestion
If only one solver submits solutions, $$\textrm{referenceScore}_i$$ is, by definition, zero.
```

fhenneke commented on docs/cow-protocol/reference/core/auctions/rewards.md:
```diff
@@ -32,48 +32,50 @@ Here $$\textrm{totalScore}$$ is the sum of the scores of all winning solutions i
+### Buffer accounting
 
-In addition to paying for gas, the winning solver might incur additional costs, such as, for example, negative slippage once a solution is settled on chain. These costs are not an explicit element of the mechanism, but they are relevant in determining the solver's optimal strategy. More precisely, per [CIP-17](https://snapshot.org/#/cow.eth/proposal/0xf9c98a2710dc72c906bbeab9b8fe169c1ed2e9af6a677 ...
+It is possible that the state of the external liquidity sources used as part of a solution changes between the bidding stage and the execution stage. We say that there is negative slippage when the solver's execution generates less than the expected amount, and positive slippage otherwise. In case of negative slippage the solver can utilize the buffers available in the settlement contract, while  ...
```
This section reads a lot better now.

I would probably remove the reference to CIP-17 as there are actually way more CIPs talking about buffer accounting. The details can then be collected in the accounting section itself.

fhenneke replied:
```suggestion
It is possible that the state of the external liquidity sources used as part of a solution changes between the bidding stage and the execution stage. We say that there is negative slippage when the solver's execution generates less than the expected amount, and positive slippage otherwise. In case of negative slippage the solver can utilize the buffers available in the settlement contract, while a solver can accumulate positive slippage in the settlement contract. Every week, the protocol computes a "net" slippage and either pays a solver (when it is positive) or requests a payment from a solver (when it is negative). For more information, see the [accounting section](/cow-protocol/reference/core/auctions/accounting).
```

fhenneke commented on docs/cow-protocol/reference/core/auctions/rewards.md:
I typically use the term "routing" for single orders only. Would one use that for batches as well?

fhenneke commented on docs/cow-protocol/reference/core/auctions/rewards.md:
Leaving the fraction of capped auctions in the docs is a bit risky as it can change at any time. I would either add a date ("(as of November 2025)") or remove the concrete number.

acanidio-econ replied:
Good point, I changed the text

fhenneke reviewed (approved):
I just added minor comments. Otherwise looks good to me.

acanidio-econ committed:
Update docs/cow-protocol/reference/core/auctions/rewards.md

Co-authored-by: Felix Henneke <felix.henneke@protonmail.com>

acanidio-econ committed:
Update docs/cow-protocol/reference/core/auctions/rewards.md

Co-authored-by: Felix Henneke <felix.henneke@protonmail.com>

acanidio-econ committed:
Update rewards.md

added "at the time of writing (Nov 2025)" when mentioning that only 9% of auctions are capped

fhenneke committed:
Merge branch 'main' into acanidio-econ-patch-12
